### PR TITLE
add module base32 encoder 

### PR DESCRIPTION
**Pull request by ismailtasdelen**
Tell us what this change does. If you're fixing a bug, please mention
the github issue number.

## Verification

List the steps needed to make sure this thing works

- [x] Start `msfconsole`
- [x] `use use auxiliary/scanner/ftps/ftps_version`
- [x] ...
- [x] **Verify** the thing does what it should
- [x] **Verify** the thing does not do what it should not
- [x] **Document** the thing and how it works ([Example](https://github.com/rapid7/metasploit-framework/blob/master/documentation/modules/post/multi/gather/aws_keys.md))

### CODE REVIEW

**Comment by ismailtasdelen** [comment]
The scanner module I have prepared is running.

downloadable module :ok_hand:  --> [ftps_version.zip](https://github.com/rapid7/metasploit-framework/files/2545690/ftps_version.zip)

**Comment by bcoles** [comment]
Does the existing FTP version scanner module not work for FTPS servers?

I'm also kind of confused about the `Msf::Exploit::Remote::Ftps` mixin included in your module. This mixin doesn't exist.

**Comment by ismailtasdelen** [comment]
Hello @bcoles 

I'm writing a metasploit module for the first time. It differs from the FTP scanner module. FTPS is a different protocol.

Msf::Exploit::Remote::Ftps --> You're right about that. I fixed it.

Thanks

**Comment by ismailtasdelen** [comment]
Hello @bcoles ,

I saw him involved.

Thanks

**Comment by h00die** [comment]
What @bcoles is referring to is that SSL is often obfuscated from the user.  For instance, most HTTP modules work seamlessly with HTTPS as well, just by setting the different port.
Most likely `auxiliary/scanner/ftp/ftp_version` set to port 990 (or 21 if it wasn't changed) may also work against ftps, and therefore a new module isn't required.

**Comment by ismailtasdelen** [comment]
I can't see it when I look at the master. I'm guessing it wasn't a master. When is it included?

**Comment by h00die** [comment]
Shodan showed me some servers claiming FTPS on 21.
```
msf auxiliary(scanner/ftp/ftp_version) > set rhosts 132.210.236.33
rhosts => 132.210.236.33
msf auxiliary(scanner/ftp/ftp_version) > run

[*] 132.210.236.33:21     - Connecting to FTP server 132.210.236.33:21...
[*] 132.210.236.33:21     - Connected to target FTP server.
[+] 132.210.236.33:21     - FTP Banner: '220 Serveur FTPS pr\xc3\xaat.\x0d\x0a'
[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
```
and nmap
```
PORT   STATE SERVICE REASON         VERSION
21/tcp open  ftp     syn-ack ttl 48 ProFTPD (requires SSL)

```

**Comment by ismailtasdelen** [comment]
Thanks for the came back :+1:

**Comment by h00die** [comment]
had to switch between several computers :)

**Comment by ismailtasdelen** [comment]
Sorry :D

File: modules/encoders/ruby/md5.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Encoder
+  Rank = GreatRanking
+
+  def initialize
+    super(
+      'Name'             => 'Ruby Md5 Encoder',
```
Comment (bcoles) [inline modules/encoders/ruby/md5.rb]: MD5 is a one-way hashing algorithm. I'm not sure that it would be useful as an encoder.

Comment (ismailtasdelen) [reply modules/encoders/ruby/md5.rb]: You're right. So I made changes to md5. I set the md5 algorithm to base32.

**Comment by bcoles** [comment]
Hi @ismailtasdelen 

Looks like `msftidy` is complaining.

```
[*] Running msftidy.rb in ./.git/hooks/post-merge mode
--- Checking new and changed module syntax with tools/dev/msftidy.rb ---
modules/encoders/ruby/base32.rb - [INFO] No CVE references found. Please check before you land!
modules/encoders/ruby/base32.rb:19 - [WARNING] Space-Tab mixed indent: "\t ['URL', 'https://www.linkedin.com/in/ismailtasdelen/'],\n"
modules/encoders/ruby/base32.rb:19 - [WARNING] Tabbed indent: "\t ['URL', 'https://www.linkedin.com/in/ismailtasdelen/'],\n"
------------------------------------------------------------------------
------------------------------------------------------------------------
[*] This merge contains modules failing msftidy.rb
[*] Please fix this if you intend to publish these
[*] modules to a popular metasploit-framework repo
------------------------------------------------------------------------
```

Also, it is required that code in your fork be merged **from a unique branch** in your repository to master in Rapid7's.  Please create a new branch in your fork of framework and resubmit this from that branch.

```
git checkout -b <BRANCH_NAME>
git push <your_fork_remote> <BRANCH_NAME>
```

This helps protect the process, ensure users are aware of commits on the branch being considered for merge, allows for a location for more commits to be offered without mingling with other contributor changes and allows contributors to make progress while a PR is still being reviewed.

Closing based on the this requirement.

File: modules/encoders/ruby/md5.rb (hunk 1 above)
Comment (wvu-r7) [reply modules/encoders/ruby/md5.rb]: > I set the md5 algorithm to base32.

You what.

**Pull request closed without merging**